### PR TITLE
#182282571: MODELS.PY TEST

### PR DESCRIPTION
**Pull request by MwanikiN**
#### What does this Pull Request do?
. Creates tests for models.py file
#### Description of Task to be completed?
. Creates tests for models.py file
#### How should this be manually tested?
. Pytest should return all successful compilations. It should also highlight the missing tests
#### Any background context you want to provide?
. Tests are fundamental during the development process as they aid in fixing bugs as well as catching impending errors
#### What are the relevant Pivotal tracker card?
. [Start django app - users](https://www.pivotaltracker.com/n/projects/2570604)
#### Screenshots (if appropriate)
. 
image

#### Questions:
N/A

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+asgiref = "==3.5.2"
```
Comment (mainamuriuki274) [inline Pipfile]: why do you have so many unnecessary packages, remove the ones you do not need and also add black and other formatters to dev dependencies

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+sqlparse = "==0.4.2"
+toml = "==0.10.2"
+tomli = "==2.0.1"
+typing-extensions = "==4.2.0"
+tzdata = "==2022.1"
+virtualenv = "==20.14.1"
+djangorestframework-simplejwt = "*"
+
+[dev-packages]
+pytest-django = "*"
```
Comment (mainamuriuki274) [inline Pipfile]: add the package version used

File: core/urls.py [hunk 3]
```diff
@@ -21,3 +21,7 @@
     path("admin/", admin.site.urls),
     path("api/v1/auth/", include("users.urls")),
 ]
+
+if os.getenv("ENV_NAME") == "local":
+        urlpatterns += static(settings.MEDIA_URL,
+                              document_root=settings.MEDIA_ROOT)
```
Comment (mainamuriuki274) [inline core/urls.py]: add an empty line after this file

File: users/models.py [hunk 4]
```diff
@@ -1,64 +1,60 @@
...
+        if not password:
+            raise ValueError("Password is required")
+        if kwargs.get("is_staff") is not True:
+            raise ValueError("Superuser must have is_staff (is_staff= True)")
+        if kwargs.get("is_superuser") is not True:
+            raise ValueError("Superuser must have is_superuser(is_superuser = True)")
+        return self.create_user(email, password, **kwargs)
+    
+
+class User(AbstractBaseUser, PermissionsMixin):
```
Comment (mainamuriuki274) [inline users/models.py]: Why have you removed  code written by Brion, e,g TimeStampedModel from inherited classes

File: users/models.py [hunk 5]
```diff
@@ -1,64 +1,60 @@
...
+
+        objects = UserManager()
+        
+        USERNAME_FIELD= 'email' 
+        REQUIRED_FIELDS= ['username']
+
+        def __str__(self) -> str:
+            return self.email
+    
+class Profile(models.Model):
```
Comment (mainamuriuki274) [inline users/models.py]: Why have you removed timestampedmodel

File: users/test/test_models.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.test import Client
+
+
+Member = get_user_model()
```
Comment (mainamuriuki274) [inline users/test/test_models.py]: Change Member to User

File: users/test/test_models.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            email="test@test.com", password="hello"
+        )
+        self.assertEqual(user.email, "test@test.com")
+        
+    
+    def test_password_exception_raised(self):
+           with self.assertRaises(ValueError):
+            Member.objects.create_user(
+                username='me',
+                email = 'me@test.com',
```
Comment (mainamuriuki274) [inline users/test/test_models.py]: use faker library to create fake data

File: users/test/test_models.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    def test_user(self):
+        self.user = Member.objects.create(username= 'me', email = 'me@test.com', password= '12345')
+        self.user_check = Member.objects.filter(email__iexact='me@test.com')
+        self.user_exist = self.user_check.exists()
+        self.assertEqual(self.user_exist,1)
+        # self.assertRegex(user.email, str('@'))
+        self.assertIsInstance(self.user.email, str)
+      
+   
+    # def test_user(self):
```
Comment (mainamuriuki274) [inline users/test/test_models.py]: remove unused comments

File: users/test/test_models.py [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    )
+
+
+class TestUser(TestCase):
+    def test_user(self):
+        self.user = Member.objects.create(username= 'me', email = 'me@test.com', password= '12345')
+        self.user_check = Member.objects.filter(email__iexact='me@test.com')
+        self.user_exist = self.user_check.exists()
+        self.assertEqual(self.user_exist,1)
+        # self.assertRegex(user.email, str('@'))
```
Comment (mainamuriuki274) [inline users/test/test_models.py]: remove unused comments

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: users/test/test_models.py (hunk 6 above)
Comment (MwanikiN) [reply users/test/test_models.py]: Done

File: users/models.py (hunk 5 above)
Comment (MwanikiN) [reply users/models.py]: Restored.

File: users/models.py (hunk 4 above)
Comment (MwanikiN) [reply users/models.py]: Incoming changes have been accepted

File: core/urls.py (hunk 3 above)
Comment (MwanikiN) [reply core/urls.py]: Done

File: Pipfile (hunk 2 above)
Comment (MwanikiN) [reply Pipfile]: Added

File: Pipfile (hunk 1 above)
Comment (MwanikiN) [reply Pipfile]: I'm not clear on this. I already have black, flake, isort, mypy installed.

File: users/test/test_models.py (hunk 7 above)
Comment (MwanikiN) [reply users/test/test_models.py]: Done

File: users/models.py [hunk 10]
```diff
@@ -61,4 +61,4 @@ def __str__(self) -> str:
 class Profile(TimeStampedModel):
     user = models.OneToOneField(User, on_delete=models.CASCADE)
     bio = models.TextField(max_length=500, blank=True)
-    image = models.ImageField(upload_to="profile_pics", blank=True)
+    image = models.ImageField(upload_to="profile_pics", blank=True)
```
Comment (mainamuriuki274) [inline users/models.py]: Add empty line after this file

File: Pipfile [hunk 11]
```diff
@@ -0,0 +1,48 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+asgiref = "==3.5.2"
+black = "==22.3.0"
+cfgv = "==3.3.1"
```
Comment (mainamuriuki274) [inline Pipfile]: This pipfile still has issues I would advice to remove all packages and dev packages and do `pipenv install django djangorestframework psycopg2-binary python-dotenv djangorestframework-simplejwt` and `pipenv install --dev black mypy django-stubs flake8 isort faker pre-commit pytest pytest-django pytest-cov`

File: core/urls.py [hunk 12]
```diff
@@ -21,3 +21,8 @@
     path("admin/", admin.site.urls),
     path("api/v1/auth/", include("users.urls")),
 ]
+
+if os.getenv("ENV_NAME") == "local":
```
Comment (mainamuriuki274) [inline core/urls.py]: Why the if , set static root in settings

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: users/models.py (hunk 10 above)
Comment (MwanikiN) [reply users/models.py]: Done

File: core/urls.py (hunk 12 above)
Comment (MwanikiN) [reply core/urls.py]: Done

File: Pipfile (hunk 11 above)
Comment (MwanikiN) [reply Pipfile]: Done

**Review by mainamuriuki274 — APPROVED** [review]

**Pull request merged**